segmentation_models_pytorch/base/model.py

```diff
@@ -137,3 +137,29 @@ def load_state_dict(self, state_dict, **kwargs):
                 warnings.warn(text, stacklevel=-1)
 
         return super().load_state_dict(state_dict, **kwargs)
+
+    def encoder_freeze(self):
+        """
+        Freeze the encoder parameters and normalization layers.
+
+        This method sets ``requires_grad = False`` for all encoder parameters,
+        preventing them from being updated during backpropagation. In addition,
+        it switches BatchNorm and InstanceNorm layers into evaluation mode
+        (``.eval()``), which stops updates to their running statistics
+        (``running_mean`` and ``running_var``) during training.
+
+        **Important:** If you call :meth:`model.train()` after
+        :meth:`encoder_freeze`, the encoder’s BatchNorm/InstanceNorm layers
+        will be put back into training mode, and their running statistics
+        will be updated again. To re-freeze them after switching modes,
+        call :meth:`encoder_freeze` again.
```

Comment:
To make the user's experience better, we can override `train` method for the model similar to the [original](https://github.com/pytorch/pytorch/blob/ba56102387ef21a3b04b357e5b183d48f0afefc7/torch/nn/modules/module.py#L2867) one

```py
    def train(self, mode: bool = True) -> Self:
        if not isinstance(mode, bool):
            raise ValueError("training mode is expected to be boolean")
        self.training = mode
        for name, module in self.named_children():
            # skip encoder encoder if it is frozen
            if self._is_encoder_frozen and name == "encoder":
                continue
            module.train(mode)
        return self
```